aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -207,9 +224,9 @@ def __init__(self, response: Response, route: Optional[Route] = None):
         self.response = response
         self.route = route
 
-    def _add_cors(self, cors: CORSConfig):
+    def _add_cors(self, event: BaseProxyEvent, cors: CORSConfig):
         """Update headers to include the configured Access-Control headers"""
-        self.response.headers.update(cors.to_dict())
+        self.response.headers.update(cors.to_dict(event.get_header_value("Origin")))
```

Comment:
Hard to see in mobile but perhaps at this point the self.current_event is already populated so you wouldn't need to expand the signature to get the event.

Unless you want to make it more explicit or future proof to work in containers next year which is sensible too

Reply:
I prefer to make it explicit, and since this a private method, I think it's ok to change the signature. Is that ok?